geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/rest/DeploymentManagementRedployDUnitTest.java

```diff
@@ -134,14 +135,16 @@ public void redeployJarsWithNewVersionsOfFunctionsAndMultipleLocators() {
     server.invoke(() -> assertThatCanLoad(JAR_NAME_A, FUNCTION_A));
     server.invoke(() -> assertThatFunctionHasVersion(FUNCTION_A, VERSION1));
 
-
     deployment.setFile(jarAVersion2);
     assertManagementResult(client.create(deployment)).isSuccessful();
     server.invoke(() -> assertThatCanLoad(JAR_NAME_A, FUNCTION_A));
     server.invoke(() -> assertThatFunctionHasVersion(FUNCTION_A, VERSION2));
 
     server.stop(false);
 
+    // GEODE-7902: wait for a while for new jars to settle in the file system
```

Comment:
What is the sleep waiting for? What does "settle" mean? Seems like whatever it means couldn't we then do a file system await to figure out if the file has settled?

Reply:
Would an `awaitUntilAsserted()` around the final assertion (or the final two) suffice, or is it necessary to wait (for whatever the condition is) before doing the restart?

If `awaitUntilAsserted()` works, that's a much more robust fix than a fixed-duration sleep.

Reply:
I believe the wait has to be before the server restarts to get the new version of jars from one of the locators.